### PR TITLE
Gate `liboxen::test` behind `test-utils` feature

### PR DESCRIPTION
**Pull request by CleanCut**
- Gate the `test` helper module behind a new `test-utils` cargo feature so release builds stop compiling/shipping it. 
- We use dev-dependencies to enable the `test-utils` feature for the other crates during tests and benchmarks.
- I moved `REPO_ROOT` and `TEST_DATA_DIR` to a small `crate::test_paths` module so the dev-mode `TEST` env-var branches in `AuthConfig::get` / `UserConfig::get` keep working in release builds.
  - I plan to refactor this small module away in a follow-up PR to add support for specifying the config directory more explicitly.

### CODE REVIEW

**Comment by malcolmgreaves** [comment]
Is a new feature needed because these things are also used in benchmarks?

**Comment by CleanCut** [comment]
> Is a new feature needed because these things are also used in benchmarks?

Good question--I dislike adding new features, but it's difficult to avoid in this scenario.

I tried using `cfg`. The problem is that while the test code resides in liboxen, it is used by the cli and server. When we run tests in the cli/server, the test profile applies to them, specifically, not dependencies. So liboxen doesn't get compiled in test mode. So the test code is unavailable if we hide it behind `cfg`. 🤦🏻‍♂️ 

We could move the test code to a separate crate that everything depends on. That seemed like a bigger change, though, and my goal was to clear the way for adding a way to specify the config directory without using the TEST env var. Besides, if we're going to put it more effort...

We could also dig into all the test code and try to move/duplicate the test code in each place that uses it. That might be the best approach overall, but because with the amount of code (~2500 lines) I thought that would take a few days, and it didn't feel justified taking the time at the moment.

Thoughts?

File: crates/lib/src/lib.rs [hunk 1]
```diff
@@ -72,7 +72,8 @@ pub mod repositories;
 pub mod request_context;
 pub mod resource;
 pub mod storage;
+#[cfg(any(test, feature = "test-utils"))]
+pub mod test;
+pub mod test_paths;
```
[the remark this answers is not in the document]
Comment (CleanCut) [reply crates/lib/src/lib.rs:77]: That's okay. I'm taking care of that in the next PR.

**Review by malcolmgreaves — APPROVED** [review]

**Comment by malcolmgreaves** [comment]
Got it. I think having the `test-utils` feature is fine. We can avoid bundling test-only code in `liboxen` (_a win IMO!_). 

> We could move the test code to a separate crate that everything depends on.

This would be lovely! One problem I ran into when I tried to do this is that this test code cannot have any dependency on `liboxen`. Unfortunately, `liboxen` will be compiled twice in such a setup: one for the non-test code and one for test code. When I tried this, I ran into the error that type/struct/enum/etc. definitions would be duplicated. A whole lot of head-scratching compiler errors where one `liboxen::error::OxenError` wouldn't match another `liboxen::error::OxenError`. 

I do think that the test utilities could have a overhaul. But IMO the biggest win is avoiding including test-only code in `liboxen`. There's a few things that are awkward with testing because of code design. The one that comes to mind is  how oxen's configs' `new` doesn't take a path. So they're less flexible and they need this odd check _internal_ to the function on the `TEST` env var being present. 😬 So overall, I am thinking about how we can improve things with respect to testing.

**Comment by CleanCut** [comment]
> The one that comes to mind is how oxen's configs' `new` doesn't take a path. So they're less flexible and they need this odd check _internal_ to the function on the `TEST` env var being present. 😬 So overall, I am thinking about how we can improve things with respect to testing.

Exactly! I already had the changes to fix that all prepped and ready for after I merge this PR.

**Comment by CleanCut** [comment]
https://github.com/Oxen-AI/Oxen/pull/475
